docs/DeveloperGuide.md

```diff
@@ -532,133 +535,213 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+1. User requests to add a friend and specifies a `FRIEND_ID` and optionally specifies a `NAME`.
 2. gitGud adds a friend into the friends list and informs user of successful addition of friend.
 
     Use case ends.
 
 **Extensions** 
 
 * 1a. gitGud detects that the `FRIEND_ID` provided already exists and is not unique.
-    * 1a1. gitGud informs user that it has failed to add a friend.<br>
-      1a2. User makes another request to add a friend with a different `FRIEND_ID`.<br>
-      Use case continues from step 2.
+    * 1a1. gitGud informs user that it has failed to add a friend.
+    * 1a2. User makes another request to add a friend with a different valid and unique `FRIEND_ID`.
+    * Use case continues from step 2.
+
+* 1b. gitGud detects that the `FRIEND_ID`/`NAME` provided is invalid and does not meet specifications provided in User Guide.
+    * 1b1. 1b1. gitGud informs user that `FRIEND_ID`/`NAME` provided is invalid.
+    * 1b2. User makes another request to add a friend with a different valid `FRIEND_ID`/`NAME` and unique `FRIEND_ID`.
+    * Use case continues from step 2.
 
-**Use case: UC02 - Delete a friend**
+**Use case: UC02 - Edit a friend**
 
 **MSS**
 
-1. User requests to delete a friend using the friend's `FRIEND_ID`
-2. gitGud deletes the person
+1. User requests to edit a friend and specifies a `FRIEND_ID` and new `NAME` to update.  
+2. gitGud updates friend and informs user of successful edit of friend.  
 
-    Use case ends.
+   Use case ends.  
 
 **Extensions**
 
-* 2a. The given `FRIEND_ID` is invalid.
+* 1a. gitGud detects that the `FRIEND_ID` does not exist.  
+    * 1a1. gitGud informs user that it has no friend was found with specified `FRIEND_ID`.  
+    * 1a2. User makes another request to edit a friend with a different `FRIEND_ID` for an existing friend.  
+    * Use case continues from step 2.  
+
+* 1b. gitGud detects that the `NAME` provided is invalid and does not meet specifications provided in User Guide.
+    * 1b1. gitGud informs user that `NAME` provided is invalid.
+    * 1b2. User makes another request to edit friend with `FRIEND_ID` with a different valid `NAME`.
+    * Use case continues from step 2.
+
+**Use case: UC03 - Delete a friend**
+
+**MSS**
+
+1. User requests to delete a friend using a friend's `FRIEND_ID`.  
+2. gitGud deletes the friend from friend list and informs user of successful deletion of friend.  
 
-    * 2a1. gitGud shows an error message.
+    Use case ends.
 
-      Use case resumes at step 1.
+**Extensions**
 
+* 1a. gitGud detects that the `FRIEND_ID` does not exist.  
+    * 1a1. gitGud informs user that it has no friend was found with specified `FRIEND_ID`.  
+    * 1a2. User makes another request to delete a friend with a different `FRIEND_ID` for an existing friend.
+    * Use case continues from step 2.  
 
-**Use case: UC03 - List friends whose `FRIEND_ID` contains a keyword**
+**Use case: UC04 - Link a game to a friend**
```

Comment:
```suggestion
**Use case: UC04 - Link a friend to a game**
```
This is really minor but below it's written as friend to a game.

Reply:
Good point, Ill keep it as link game to friend as that is how it is stated in the UG, but instead change the bottom line to link game to friend as well for consistency.

Reply:
Sure 👍🏽